src/compiler/binder.ts

```diff
@@ -106,6 +110,27 @@ namespace ts {
         return ModuleInstanceState.Instantiated;
     }
 
+    function isExportedInBlock(name: string, statement: Node): boolean {
+        switch (statement.kind) {
+            case SyntaxKind.ModuleDeclaration:
+                const decl = statement as ModuleDeclaration;
+                return !!decl.body && isModuleBlock(decl.body) && some(decl.body.statements, s => isExportedInBlock(name, s));
+            case SyntaxKind.SourceFile:
+            case SyntaxKind.Block:
+                return some((statement as Block | SourceFile).statements, s => isExportedInBlock(name, s));
+            case SyntaxKind.ExportDeclaration:
+                if (isExportDeclaration(statement) && !statement.moduleSpecifier && statement.exportClause && isNamedExports(statement.exportClause)) {
+                    for (const spec of statement.exportClause.elements) {
+                        if (name === idText(spec.propertyName || spec.name)) {
+                            return true;
+                        }
+                    }
+                }
+                break;

```

Comment:
Maybe a `return false;` would be more readable, following the other cases?  (Also if it's extended later.)